akka-actor/src/main/scala/akka/io/dns/internal/DnsMessage.scala

```diff
@@ -135,10 +135,10 @@ private[internal] object Message {
     val nsCount = it.getShort
     val arCount = it.getShort
 
-    val qs = (0 until qdCount) map { i ⇒ Try(Question.parse(it, msg)) }
-    val ans = (0 until anCount) map { i ⇒ Try(ResourceRecord.parse(it, msg)) }
-    val nss = (0 until nsCount) map { i ⇒ Try(ResourceRecord.parse(it, msg)) }
-    val ars = (0 until arCount) map { i ⇒ Try(ResourceRecord.parse(it, msg)) }
+    val qs = (0 until qdCount) map { _ ⇒ Try(Question.parse(it, msg)) }
```

Comment:
Maybe 'foreach' instead of 'map'?

Reply:
The val is needed 
```scala
new Message(id, flags, qs.flatten, ans.flatten, nss.flatten, ars.flatten)
```
but your suggestions return Unit
```scala
val a: IndexedSeq[Try[ResourceRecord] =
  (0 until arCount).map { _ ⇒ Try(ResourceRecord.parse(it, msg)) }

val b: Unit =
  (0 until arCount.foreach { _ ⇒ Try(ResourceRecord.parse(it, msg)) }
```

Reply:
ok, I guess we didn't looked close enough how it was used

Reply:
Ah you're right of course. `List.fill(arCount) { Try(ResourceRecord.parse(it, msg)) }`? Though I agree it's less obvious that it's an improvement ;)